src/common/collection.c

```diff
@@ -1830,14 +1830,14 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
       if(strcmp(operator, "[]") == 0)
       {
         if(number1 && number2)
-          query = g_strdup_printf("((%s >= %ld) AND (%s <= %ld))", colname, (long int)nb1, colname, (long int)nb2);
```

Comment:
You have to change the format too, %ld is wrong. I think you have to use G_GUINT64_FORMAT

Reply:
Or rather G_GINT64_FORMAT

Reply:
Or, as alternative to GLib types and macros, use inttypes.h standard/portable width integer types and macros.

Reply:
I would agree, but we already stuck with glib, since nb1 type is GTimeSpan, which is defined as gint64: https://developer-old.gnome.org/glib/stable/glib-GDateTime.html#GTimeSpan

So it's better to match formatting macros to the actual type, even if it's all the same underneath.

Reply:
I had a doubt about %ld indeed. Thanks for your suggestions. Done.